telegram/ext/updater.py

```diff
@@ -475,6 +527,23 @@ def bootstrap_clean_updates():
                 updates = self.bot.get_updates(updates[-1].update_id + 1)
             return False
 
+        def bootstrap_clean_updates_datetime(datetime_cutoff):
+            self.logger.debug('Cleaning updates from Telegram server with datetime "%s"',
+                                    datetime_cutoff)
+            updates = self.bot.get_updates()
+
+            # reversed as we just need to find the first msg that's too old
+            for up in reversed(updates):
+                self.logger.debug('cutoff:              "%s"', datetime_cutoff)
+                self.logger.debug('msg date:            "%s"', up.message.date)
+                self.logger.debug('msg date < cutoff:   "%s"', (up.message.date < datetime_cutoff))
+                if up.message and (up.message.date < datetime_cutoff):
```

Comment:
this needs to be `effective_message` as `channel_post`, and `edited_*` are messages as well

Reply:
Good point

Reply:
resolved with cb7bb3b

Reply:
Doesn't cover `edited_*`, though